src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/WorkspaceProject.cs

```diff
@@ -144,18 +144,17 @@ public async Task SetBuildSystemPropertiesAsync(IReadOnlyDictionary<string, stri
                 case "TargetPath": _project.OutputFilePath = GetFullyQualifiedPath(valueOrNull); break;
                 case "TargetRefPath": _project.OutputRefFilePath = GetFullyQualifiedPath(valueOrNull); break;
                 case "TargetFrameworkIdentifier": _targetFrameworkManager.UpdateIdentifierForProject(_project.Id, valueOrNull); break;
+                case "IntermediateAssembly": _project.CompilationOutputAssemblyFilePath = GetFullyQualifiedPath(valueOrNull); break;
             }
         }
 
-        // Workaround for https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1830960
-        _project.CompilationOutputAssemblyFilePath = _project.OutputFilePath;
```

Comment:
This issue was fixed, and seems like my machine has the change, so I guess its been released too.

FYI @tmat

Reply:
What was OutputFilePath before, the bin path or obj path? Is this changing it?

Reply:
Yeah, it was the path to the DLL in the bin folder, so yes this changes it. Hence the Tomas tag to ensure this won't break Hot Reload :)